common/src/main/java/dev/latvian/mods/kubejs/client/SoundGenerator.java

```diff
@@ -0,0 +1,157 @@
+package dev.latvian.mods.kubejs.client;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import net.minecraft.Util;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.util.Mth;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+public class SoundGenerator {
+
+	private final Map<String, SoundEntry> entries = new HashMap<>();
+
+	public void addSound(String path, Consumer<SoundEntry> consumer) {
+		if (entries.containsKey(path)) {
+			consumer.accept(entries.get(path));
+		} else {
+			entries.put(path, Util.make(new SoundEntry(), consumer));
+		}
+	}
+
+	public JsonObject toJson() {
+		final JsonObject json = new JsonObject();
+		entries.forEach((path, entry) -> json.add(path, entry.toJson()));
+		return json;
+	}
+
+	public static class SoundEntry {
+
+		private boolean replace = false;
+		@Nullable
+		private String subtitle;
+		private final List<SoundInstance> sounds = new ArrayList<>();
+
+		public SoundEntry replace(boolean b) {
+			replace = b;
+			return this;
+		}
```

Comment:
Should probably add an overload `replace() { return replace(true) }` like other methods of this form in KJS tend to have.